src/Microsoft.AspNet.Razor.Runtime/TagHelperContent.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.Razor.Runtime
+{
+    /// <summary>
+    /// Abstract class which is used to buffer content returned by <see cref="TagHelpers.ITagHelper"/>.
+    /// </summary>
+    public abstract class TagHelperContent
+    {
+        /// <summary>
+        /// Indicates whether the content was modifed.
+        /// </summary>
+        public abstract bool IsModified { get; }
+
+        /// <summary>
+        /// Returns <c>true</c> if <see cref="TagHelperContent"/> is empty, <c>false</c> otherwise.
+        /// </summary>
+        public abstract bool IsEmpty { get; }
+
+        /// <summary>
+        /// Returns <c>true</c> if <see cref="TagHelperContent"/> is whitespace, <c>false</c> otherwise.
+        /// </summary>
+        public abstract bool IsWhiteSpace { get; }
+
+        /// <summary>
+        /// Sets the value of <see cref="TagHelperContent"/>.
+        /// </summary>
+        /// <param name="value">The value to which <see cref="TagHelperContent"/> must be set.</param>
+        public abstract void SetContent(string value);
+
+        /// <summary>
+        /// Appends to the existing value of <see cref="TagHelperContent"/>.
+        /// </summary>
+        /// <param name="value">The value which must be appended.</param>
+        public abstract void Append(string value);
```

Comment:
Part of me wants to make this a fluent API by returning `TagHelperContent` but that does make it slightly odd from our (framework devs) perspective because it makes `DefaultTagHelperContent.Append` return a `TagHelperContent` even though you're working on a `DefaultTagHelperContent` bonus though is that for users you're only ever working with `TagHelperContent` so it makes life easier.